<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to clustermole</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to clustermole</h1>



<div id="overview" class="section level2">
<h2>Overview</h2>
<p>A typical computational pipeline to process single-cell RNA sequencing (scRNA-seq) data includes clustering of cells as one of the steps. Assignment of cell type labels to those clusters is often a time-consuming process that involves manual inspection of the cluster marker genes complemented with a detailed literature search. This is especially challenging when unexpected or poorly described populations are present. The clustermole R package provides methods to query thousands of human and mouse cell identity markers sourced from a variety of databases.</p>
<p>The clustermole package provides three primary features:</p>
<ul>
<li>a database of markers for thousands of cell types (<code>clustermole_markers</code>)</li>
<li>cell type prediction based on marker genes (<code>clustermole_overlaps</code>)</li>
<li>cell type prediction based on the full expression matrix (<code>clustermole_enrichment</code>)</li>
</ul>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p>You can install clustermole from <a href="https://cran.r-project.org/package=clustermole">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;clustermole&quot;</span>)</span></code></pre></div>
<p>Load clustermole.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(clustermole)</span></code></pre></div>
<div id="retrieve-cell-type-markers" class="section level3">
<h3>Retrieve cell type markers</h3>
<p>You can use clustermole as a simple database and get a data frame of all cell type markers.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>markers &lt;-<span class="st"> </span><span class="kw">clustermole_markers</span>(<span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a>markers</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; # A tibble: 422,292 x 8</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt;    celltype_full      db     species organ  celltype   n_genes gene_origi… gene </span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt;  1 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACCSL       ACCSL</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt;  2 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ACVR1B      ACVR…</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt;  3 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 ASF1B       ASF1B</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">#&gt;  4 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BCL2L10     BCL2…</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">#&gt;  5 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 BLCAP       BLCAP</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">#&gt;  6 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CASC3       CASC3</span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co">#&gt;  7 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CLEC10A     CLEC…</span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="co">#&gt;  8 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 CNOT11      CNOT…</span></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="co">#&gt;  9 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DCLK2       DCLK2</span></span>
<span id="cb3-15"><a href="#cb3-15"></a><span class="co">#&gt; 10 1-cell stage cell… CellM… Human   Embryo 1-cell st…      32 DHCR7       DHCR7</span></span>
<span id="cb3-16"><a href="#cb3-16"></a><span class="co">#&gt; # … with 422,282 more rows</span></span></code></pre></div>
<p>Each row contains a gene and a cell type associated with it. The <code>gene</code> column is the gene symbol and the <code>celltype_full</code> column contains the full cell type string, including the species and the original database. Human or mouse versions can be retrieved.</p>
<p>Some tools require input as a list. To convert the markers from a data frame to a list format, you can use <code>gene</code> as the values and <code>celltype_full</code> as the grouping variable.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>markers_list &lt;-<span class="st"> </span><span class="kw">split</span>(<span class="dt">x =</span> markers<span class="op">$</span>gene, <span class="dt">f =</span> markers<span class="op">$</span>celltype_full)</span></code></pre></div>
<p>Check the number of cell types in the database.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">length</span>(<span class="kw">unique</span>(markers<span class="op">$</span>celltype_full))</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co">#&gt; [1] 3039</span></span></code></pre></div>
<p>Check the cell type source databases.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw">sort</span>(<span class="kw">unique</span>(markers<span class="op">$</span>db))</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="co">#&gt; [1] &quot;ARCHS4&quot;     &quot;CellMarker&quot; &quot;MSigDB&quot;     &quot;PanglaoDB&quot;  &quot;SaVanT&quot;    </span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt; [6] &quot;TISSUES&quot;    &quot;xCell&quot;</span></span></code></pre></div>
</div>
<div id="cell-types-based-on-marker-genes" class="section level3">
<h3>Cell types based on marker genes</h3>
<p>If you have a character vector of genes, such as cluster markers, you can compare them to known cell type markers to see if they overlap any of the known cell type markers (overrepresentation analysis).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>my_overlaps &lt;-<span class="st"> </span><span class="kw">clustermole_overlaps</span>(<span class="dt">genes =</span> my_genes_vec, <span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</span></code></pre></div>
</div>
<div id="cell-types-based-on-expression-matrix" class="section level3">
<h3>Cell types based on expression matrix</h3>
<p>If you have expression values, such as average expression across clusters, you can perform cell type enrichment based on a given gene expression matrix (log-transformed CPM/TPM/FPKM values). The matrix should have genes as rows and clusters/samples as columns. The underlying enrichment method can be changed using the <code>method</code> parameter.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>my_enrichment &lt;-<span class="st"> </span><span class="kw">clustermole_enrichment</span>(<span class="dt">expr_mat =</span> my_expr_mat, <span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
