<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Introduction to clustermole</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to clustermole</h1>



<p><em>Alternative title: blindly digging for cell types in scRNA-seq clusters with clustermole</em></p>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>A typical computational pipeline to process single-cell RNA sequencing (scRNA-seq) data involves clustering of cells. Assignment of cell type labels to those clusters is often a time-consuming process that involves manual inspection of the cluster marker genes complemented with a detailed literature search. This is especially challenging if you are not familiar with all the captured subpopulations or have unexpected contaminants. <code>clustermole</code> is an R package that provides a comprehensive meta collection of cell identity markers for thousands of human and mouse cell types sourced from a variety of databases as well as methods to query them.</p>
<p>The <code>clustermole</code> package includes three primary features:</p>
<ul>
<li>cell type identification based on a set of marker genes (<code>clustermole_overlaps</code>)</li>
<li>cell type identification based on a full expression matrix (<code>clustermole_enrichment</code>)</li>
<li>a meta collection of cell type markers (<code>clustermole_markers</code>)</li>
</ul>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p>Install <code>clustermole</code> if it is not yet available on your system.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="st">&quot;clustermole&quot;</span>)</a></code></pre></div>
<p>Load <code>clustermole</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(clustermole)</a></code></pre></div>
<div id="overlap-a-set-of-genes-with-cell-type-markers" class="section level3">
<h3>Overlap a set of genes with cell type markers</h3>
<p>If you have a set of genes (for example, cluster markers), you can perform overrepresentation analysis to see if they overlap any of the known cell type markers.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">my_genes =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CD2&quot;</span>, <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD3E&quot;</span>, <span class="st">&quot;IL7R&quot;</span>, <span class="st">&quot;IL32&quot;</span>, <span class="st">&quot;LTB&quot;</span>, <span class="st">&quot;LDHB&quot;</span>, <span class="st">&quot;CCR7&quot;</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">my_overlaps =<span class="st"> </span><span class="kw">clustermole_overlaps</span>(<span class="dt">genes =</span> my_genes, <span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">my_overlaps</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; # A tibble: 2,563 x 9</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;    db     species organ celltype celltype_full n_genes overlap  p_value      fdr</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;           &lt;int&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;  1 Pangl… Human   Immu… T memor… T memory cel…      54       6 3.90e-15 5.51e-12</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  2 SCSig  Human   Cent… Fan_Emb… Fan_Embryoni…     150       7 4.30e-15 5.51e-12</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;  3 Pangl… Mouse   Immu… T memor… T memory cel…      57       6 7.56e-15 6.46e-12</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;  4 CellM… Human   Peri… T cell   T cell | Per…      19       5 1.58e-14 1.01e-11</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;  5 CellM… Human   Kidn… T helpe… T helper cel…       5       4 4.06e-14 2.08e-11</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;  6 Pangl… Human   Immu… T cells  T cells | Im…      95       6 1.31e-13 5.12e-11</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  7 Pangl… Mouse   Immu… T cells  T cells | Im…      93       6 1.40e-13 5.12e-11</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  8 SaVanT &quot;&quot;      &quot;&quot;    CD3plus… CD3plus_T-ce…      50       5 2.87e-12 5.65e-10</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt;  9 SaVanT Human   &quot;&quot;    HPCA_T_… HPCA_T_cells…      50       5 2.87e-12 5.65e-10</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; 10 SaVanT Mouse   &quot;&quot;    IMGN_T_… IMGN_T_4Nve_…      50       5 2.87e-12 5.65e-10</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; # … with 2,553 more rows</span></a></code></pre></div>
</div>
<div id="determine-relative-enrichment-of-cell-type-markers-in-the-input-expression-data" class="section level3">
<h3>Determine relative enrichment of cell type markers in the input expression data</h3>
<p>If you have a table of expression values (for example, average expression across clusters), you can perform cell type enrichment based on a given gene expression matrix (log-transformed CPM/TPM/FPKM values).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">clustermole_enrichment</span>(<span class="dt">expr_mat =</span> my_expr_mat, <span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</a></code></pre></div>
</div>
<div id="retrieve-cell-type-markers" class="section level3">
<h3>Retrieve cell type markers</h3>
<p>You can retrieve a data frame of all cell type markers in the database.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">markers =<span class="st"> </span><span class="kw">clustermole_markers</span>(<span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">markers</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; # A tibble: 163,509 x 8</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;    db     species organ  celltype  celltype_full     n_genes gene_original gene </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;               &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;  1 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ACCSL         ACCSL</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;  2 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ACVR1B        ACVR…</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;  3 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ARHGEF16      ARHG…</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;  4 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ASF1B         ASF1B</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  5 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BCL2L10       BCL2…</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  6 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BLCAP         BLCAP</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  7 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BNIP1         BNIP1</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  8 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 C1orf210      C1or…</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  9 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 C1orf226      C1or…</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 10 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 CASC3         CASC3</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; # … with 163,499 more rows</span></a></code></pre></div>
<p>Each row contains a gene and a cell type associated with it. The <code>gene</code> column is the gene symbol (human or mouse versions can be retrieved) and the <code>celltype_full</code> column contains the full cell type string, including the species and the original database.</p>
<p>If you need to convert the markers from a data frame to a list format for other applications, you can use <code>gene</code> as the values and <code>celltype_full</code> as the grouping variable.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">markers_list =<span class="st"> </span><span class="kw">split</span>(<span class="dt">x =</span> markers<span class="op">$</span>gene, <span class="dt">f =</span> markers<span class="op">$</span>celltype_full)</a></code></pre></div>
</div>
</div>
<div id="collection-details" class="section level2">
<h2>Collection details</h2>
<p>We will use <code>dplyr</code> to help with summary statistics.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">library</span>(dplyr)</a></code></pre></div>
<p>Retrieve a data frame of all cell type markers in the database.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">markers =<span class="st"> </span><span class="kw">clustermole_markers</span>(<span class="dt">species =</span> <span class="st">&quot;hs&quot;</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">markers</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; # A tibble: 163,509 x 8</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt;    db     species organ  celltype  celltype_full     n_genes gene_original gene </span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;               &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;  1 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ACCSL         ACCSL</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt;  2 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ACVR1B        ACVR…</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  3 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ARHGEF16      ARHG…</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt;  4 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 ASF1B         ASF1B</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt;  5 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BCL2L10       BCL2…</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;  6 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BLCAP         BLCAP</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt;  7 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 BNIP1         BNIP1</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt;  8 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 C1orf210      C1or…</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt;  9 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 C1orf226      C1or…</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; 10 CellM… Human   Embryo 1-cell s… 1-cell stage cel…      45 CASC3         CASC3</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt; # … with 163,499 more rows</span></a></code></pre></div>
<p>Check the number of available cell types.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(celltype_full) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>()</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; [1] 2563</span></a></code></pre></div>
<p>Check the number of available cell types per species (not available for every cell type).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(celltype_full, species) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(species, <span class="dt">sort =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; # A tibble: 3 x 2</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt;   species     n</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt;   &lt;chr&gt;   &lt;int&gt;</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; 1 Human    1618</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; 2 Mouse     730</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; 3 &quot;&quot;        215</span></a></code></pre></div>
<p>Check the number of available cell types per organ (not available for every cell type).</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(celltype_full, organ) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(organ, <span class="dt">sort =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co">#&gt; # A tibble: 117 x 2</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt;    organ                      n</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt;                  &lt;int&gt;</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;  1 &quot;&quot;                      1282</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;  2 Brain                    127</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt;  3 Central Nervous System    88</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt;  4 Digestive System          63</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt;  5 Kidney                    56</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt;  6 Lung                      52</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt;  7 Bone marrow               51</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;  8 Immune system             50</span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt;  9 Peripheral blood          46</span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt; 10 Hematopoietic system      44</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt; # … with 107 more rows</span></a></code></pre></div>
<p>Check package version.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">packageVersion</span>(<span class="st">&quot;clustermole&quot;</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">#&gt; [1] '1.0.0'</span></a></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
